returns/primitives/tracing.py

```diff
@@ -1,13 +1,35 @@
 import types
 from contextlib import contextmanager
 from inspect import FrameInfo, stack
-from typing import List, Optional
+from typing import (
+    Callable,
+    ContextManager,
+    Iterator,
+    List,
+    Optional,
+    TypeVar,
+    Union,
+    overload,
+)
 
 from returns.result import _Failure
 
+Function = TypeVar('Function', bound=Callable)
```

Comment:
This should be named `_FunctionType`
1. It is protected
2. We try to use `Type` suffix to indicate that this is a typevar

Reply:
Done!